src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/EnumGroupAndName.cs

```diff
@@ -39,6 +77,53 @@ public EnumGroupAndName(string group, string name)
         /// <summary>
         /// Gets the name.
         /// </summary>
-        public string Name { get; }
+        public string Name
+        {
+            get
+            {
+                return GetDisplayName();
+            }
+        }
+
+        // Equals and GetHashCode must be overloaded to accommodate the _nameFunc
+        public override bool Equals(object obj)
+        {
+            if (!(obj is EnumGroupAndName))
+            {
+                return false;
+            }
+
+            var second = (EnumGroupAndName)obj;
+
+            return string.Equals(Group, second.Group, StringComparison.Ordinal) && string.Equals(Name, second.Name, StringComparison.Ordinal);
```

Comment:
For consistency and because it's probably better to avoid `IStringLocalizer`'s allocations, check equality of the fields.